x/protocolpool/keeper/keeper_test.go

```diff
@@ -176,78 +111,421 @@ func (suite *KeeperTestSuite) TestGetCommunityPool() {
+					Percentage: math.LegacyMustNewDecFromStr("0.3"),
+					Expiry:     &expiredTime,
+				}
+				err := suite.poolKeeper.ContinuousFunds.Set(suite.ctx, recipient, fund)
+				suite.Require().NoError(err)
+
+				// And full amount to be sent to the community pool.
+				suite.bankKeeper.EXPECT().SendCoinsFromModuleToModule(suite.ctx, types.ProtocolPoolDistrAccount, types.ModuleName, totalCoins).
+					Return(nil).Times(1)
+			},
+			expectedErr: "",
+			verify: func() {
+				// Verify that the expired fund was removed.
+				funds, err := suite.poolKeeper.GetAllContinuousFunds(suite.ctx)
+				suite.Require().NoError(err)
+				suite.Require().Empty(funds, "expected expired continuous fund to be removed")
+			},
+		},
+		{
+			name: "multiple valid continuous funds",
+			setup: func() {
+				suite.SetupTest()
+				suite.authKeeper.EXPECT().GetModuleAccount(suite.ctx, types.ProtocolPoolDistrAccount).
+					Return(poolDistrAcc).AnyTimes()
+
+				params := types.Params{EnabledDistributionDenoms: []string{sdk.DefaultBondDenom}}
+				suite.Require().NoError(suite.poolKeeper.Params.Set(suite.ctx, params))
+
+				totalCoins := sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, math.NewInt(1000)))
+				suite.bankKeeper.EXPECT().GetBalance(suite.ctx, poolDistrAcc.GetAddress(), sdk.DefaultBondDenom).
+					Return(sdk.NewCoin(sdk.DefaultBondDenom, math.NewInt(1000))).Times(1)
+
+				recipient1 := sdk.AccAddress("recipient1______________")
+				fund1 := types.ContinuousFund{
+					Recipient:  recipient1.String(),
+					Percentage: math.LegacyMustNewDecFromStr("0.3"),
+					Expiry:     nil,
+				}
+				err := suite.poolKeeper.ContinuousFunds.Set(suite.ctx, recipient1, fund1)
+				suite.Require().NoError(err)
+
+				recipient2 := sdk.AccAddress("recipient2______________")
+				fund2 := types.ContinuousFund{
+					Recipient:  recipient2.String(),
+					Percentage: math.LegacyMustNewDecFromStr("0.2"),
+					Expiry:     nil,
+				}
+				err = suite.poolKeeper.ContinuousFunds.Set(suite.ctx, recipient2, fund2)
+				suite.Require().NoError(err)
+
+				amountToStream1 := poolkeeper.PercentageCoinMul(math.LegacyMustNewDecFromStr("0.3"), totalCoins)
+				amountToStream2 := poolkeeper.PercentageCoinMul(math.LegacyMustNewDecFromStr("0.2"), totalCoins)
+				suite.bankKeeper.EXPECT().SendCoinsFromModuleToAccount(suite.ctx, types.ProtocolPoolDistrAccount, recipient1, amountToStream1).
+					Return(nil).Times(1)
+				suite.bankKeeper.EXPECT().SendCoinsFromModuleToAccount(suite.ctx, types.ProtocolPoolDistrAccount, recipient2, amountToStream2).
+					Return(nil).Times(1)
+
+				remainingCoins := totalCoins.Sub(amountToStream1...).Sub(amountToStream2...)
+				suite.bankKeeper.EXPECT().SendCoinsFromModuleToModule(suite.ctx, types.ProtocolPoolDistrAccount, types.ModuleName, remainingCoins).
+					Return(nil).Times(1)
+			},
+			expectedErr: "",
+		},
+		{
+			name: "fund percentages sum over 1 (resulting in negative remainder)",
+			setup: func() {
+				suite.SetupTest()
```

Comment:
i think we can just move the suite.SetupTest() call to before the tc.setup() call on L459.